utils/storage/storage.js

```diff
@@ -248,10 +252,45 @@ class Storage {
     );
   }
 
+  fetchAnonymousId(key) {
+    let anonId;
+    switch (key) {
+      case "segment":
+        if (Store.enabled) {
+          anonId = Store.get(anonymousIdKeyMap[key]);
+        }
+        if (!anonId) {
+          Cookie.set("rudder_cookies", true);
+
+          if (Cookie.get("rudder_cookies")) {
+            Cookie.remove("rudder_cookies");
+            anonId = Cookie.get(anonymousIdKeyMap[key]);
+          }
+        }
+        return anonId;
+
+      default:
+        return undefined;
+    }
+  }
+
   /**
    * get stored anonymous id
    */
-  getAnonymousId() {
+  getAnonymousId(anonymousIdOptions) {
+    if (
+      anonymousIdOptions &&
+      anonymousIdOptions.autoCapture &&
+      anonymousIdOptions.autoCapture.enabled &&
+      typeof anonymousIdOptions.autoCapture.source === "string"
+    ) {
+      const source = anonymousIdOptions.autoCapture.source.toLowerCase();
+      if (Object.keys(anonymousIdKeyMap).includes(source)) {
```

Comment:
These two things can go into the fetch method below.

Reply:
Done